code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsExtension.kt

```diff
@@ -375,23 +453,48 @@ internal class AnalyticsExtension(extensionApi: ExtensionApi) : Extension(extens
         trackAcquisitionData(event)
     }
 
+    /**
+     * Handler for Analytics Request Identity events.
+     * Handles public API requests to set the visitor identifier. If privacy is opted out, the
+     * request is ignored.
+     *
+     * @param event the analytics request identity event
+     */
     private fun handleAnalyticsRequestIdentityEvent(event: Event) {
-        val vid =
-            event.eventData?.get(AnalyticsConstants.EventDataKeys.Analytics.VISITOR_IDENTIFIER) as String?
-        vid?.let {
-            if (analyticsState.privacyStatus != MobilePrivacyStatus.OPT_OUT) {
-                analyticsProperties.vid = vid
-            } else {
-                Log.warning(
+        if (event.eventData?.containsKey(AnalyticsConstants.EventDataKeys.Analytics.VISITOR_IDENTIFIER) == true) {
+            if (analyticsState.privacyStatus == MobilePrivacyStatus.OPT_OUT) {
+                Log.debug(
                     AnalyticsConstants.LOG_TAG,
                     CLASS_NAME,
                     "handleAnalyticsRequestIdentityEvent - Privacy is opted out, ignoring the update visitor identifier request."
                 )
+                return
+            }
+
+            try {
+                analyticsProperties.vid = DataReader.getString(
+                    event.eventData,
+                    AnalyticsConstants.EventDataKeys.Analytics.VISITOR_IDENTIFIER
+                )
+            } catch (ex: DataReaderException) {
+                Log.debug(
+                    AnalyticsConstants.LOG_TAG,
+                    CLASS_NAME,
+                    "handleAnalyticsRequestIdentityEvent - Failed to parse the visitor identifier to string, ignoring the update visitor identifier request."
+                )
             }
         }
+
         publishAnalyticsId(event)
     }
```

Comment:
This is the only code change. It fixes a bug where a null vid value is ignored, however it should clear the existing vid value.